src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/profile/garden/PlotsAPI.kt

```diff
@@ -0,0 +1,292 @@
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyWidget
+import tech.thatgravyboat.skyblockapi.api.events.chat.ChatReceivedEvent
+import tech.thatgravyboat.skyblockapi.api.events.info.ScoreboardUpdateEvent
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidget
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidgetChangeEvent
+import tech.thatgravyboat.skyblockapi.api.events.misc.RegisterCommandsEvent
+import tech.thatgravyboat.skyblockapi.api.events.screen.InventoryChangeEvent
+import tech.thatgravyboat.skyblockapi.api.location.SkyBlockIsland
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import tech.thatgravyboat.skyblockapi.helpers.McPlayer
+import tech.thatgravyboat.skyblockapi.helpers.McPlayer.contains
+import tech.thatgravyboat.skyblockapi.impl.tagkey.ItemTag
+import tech.thatgravyboat.skyblockapi.utils.extentions.*
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexGroup
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.anyMatch
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.match
+import tech.thatgravyboat.skyblockapi.utils.regex.matchWhen
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.Text.sendWithPrefix
+import kotlin.math.floor
+
+@Module
+object PlotAPI {
+    //region Regex
+    private val inventoryGroup = RegexGroup.INVENTORY.group("plots")
+    private val scoreboardGroup = RegexGroup.SCOREBOARD.group("plots")
+    private val tablistGroup = RegexGroup.TABLIST.group("plots")
+    private val chatGroup = RegexGroup.CHAT.group("plots")
+
+    private val deskPlotNameRegex = inventoryGroup.create(
+        "name",
+        "Plot - (?<name>.+)",
+    )
+
+    private val deskPestsRegex = inventoryGroup.create(
+        "pests",
+        "ൠ This plot has (?<amount>\\d+) Pests?!",
+    )
+
+    private val scoreboardPestAmountRegex = scoreboardGroup.create(
+        "pest_amount",
+        "⏣ The Garden ൠ x(?<amount>\\d+)",
+    )
+
+    private val scoreboardNoPestsRegex = scoreboardGroup.create(
+        "no_pests",
+        " ⏣ (?:The Garden|Plot - .+)",
+    )
+
+    private val scoreboardPlotPestAmountRegex = scoreboardGroup.create(
+        "plot_pest_amount",
+        " {3}Plot - (?<name>.+) ൠ x(?<amount>\\d+)",
+    )
+
+    private val tablistAliveRegex = tablistGroup.create(
+        "pests_alive",
+        "\\s*Alive: (?<amount>\\d+)",
+    )
+
+    private val tablistPlotsRegex = tablistGroup.create(
+        "plots",
+        "\\s*Plots: (?<plots>(?:\\d+,?\\s*)+)",
+    )
+
+    private val chatSingularSpawnRegex = chatGroup.create(
+        "singular_spawn",
+        ".*! A Pest has appeared in Plot - (?<name>.+)!",
+    )
+    private val chatPluralSpawnRegex = chatGroup.create(
+        "plural_spawn",
+        ".*! (?<amount>\\d+) Pests have spawned in Plot - (?<name>.+)!",
+    )
+
+    //endregion
+
+    private var slot = 1
+
+    val plots = listOf(
```

Comment:
this is not very easily followable, can this be changed to something more readable :pleading_face:

Reply:
I think it's pretty clear what it does, it's just how the ids are arranged, don't see a way how else you can represent this that isn't way too over complicated 😭